hack/smoke-test.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+#-----------------------------------------------------------------------------
+# Global Variables
+#-----------------------------------------------------------------------------
+export V_FLAG=-v
+export OUTPUT_DIR=$(pwd)/_output
```

Comment:
running `shellcheck` on your script here produced these results (install `shellcheck` if you do not have it and address theses): 

```
$ shellcheck smoke-test.sh 

In smoke-test.sh line 7:
export OUTPUT_DIR=$(pwd)/_output
       ^--------^ SC2155: Declare and assign separately to avoid masking return values.


In smoke-test.sh line 15:
python3 -m venv ${PYTHON_VENV_DIR}
                ^----------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
python3 -m venv "${PYTHON_VENV_DIR}"


In smoke-test.sh line 16:
${PYTHON_VENV_DIR}/bin/pip install --upgrade setuptools
^----------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
"${PYTHON_VENV_DIR}"/bin/pip install --upgrade setuptools


In smoke-test.sh line 17:
${PYTHON_VENV_DIR}/bin/pip install --upgrade pip
^----------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
"${PYTHON_VENV_DIR}"/bin/pip install --upgrade pip


In smoke-test.sh line 19:
echo "test-namespace-$(uuidgen | tr '[:upper:]' '[:lower:]' | head -c 8)" > ${OUTPUT_DIR}/test-namespace
                                                                            ^-----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
echo "test-namespace-$(uuidgen | tr '[:upper:]' '[:lower:]' | head -c 8)" > "${OUTPUT_DIR}"/test-namespace


In smoke-test.sh line 20:
if [[ $OPENSHIFT_CI == true ]]; then
   ^-------------------------^ SC3010: In POSIX sh, [[ ]] is undefined.


In smoke-test.sh line 22:
    oc project -q > ${OUTPUT_DIR}/test-namespace
                    ^-----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
    oc project -q > "${OUTPUT_DIR}"/test-namespace


In smoke-test.sh line 24:
export TEST_NAMESPACE=$(cat ${OUTPUT_DIR}/test-namespace)
       ^------------^ SC2155: Declare and assign separately to avoid masking return values.
                            ^-----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
export TEST_NAMESPACE=$(cat "${OUTPUT_DIR}"/test-namespace)


In smoke-test.sh line 28:
kubectl delete namespace ${TEST_NAMESPACE} --timeout=45s --wait
                         ^---------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
kubectl delete namespace "${TEST_NAMESPACE}" --timeout=45s --wait


In smoke-test.sh line 29:
kubectl create namespace ${TEST_NAMESPACE}
                         ^---------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
kubectl create namespace "${TEST_NAMESPACE}"


In smoke-test.sh line 31:
mkdir -p ${LOGS_DIR}/smoke-tests-logs
         ^---------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
mkdir -p "${LOGS_DIR}"/smoke-tests-logs


In smoke-test.sh line 32:
mkdir -p ${OUTPUT_DIR}/smoke-tests-output
         ^-----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
mkdir -p "${OUTPUT_DIR}"/smoke-tests-output


In smoke-test.sh line 33:
touch ${OUTPUT_DIR}/backups.txt
      ^-----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
touch "${OUTPUT_DIR}"/backups.txt


In smoke-test.sh line 35:
echo "Logs directory created at "${LOGS_DIR}/smoke
                                 ^---------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
echo "Logs directory created at ""${LOGS_DIR}"/smoke


In smoke-test.sh line 38:
oc project ${TEST_NAMESPACE}
           ^---------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
oc project "${TEST_NAMESPACE}"


In smoke-test.sh line 42:
${PYTHON_VENV_DIR}/bin/pip install -q -r smoke/requirements.txt
^----------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
"${PYTHON_VENV_DIR}"/bin/pip install -q -r smoke/requirements.txt


In smoke-test.sh line 44:
echo "Logs will be collected in "${TEST_SMOKE_OUTPUT_DIR}
                                 ^----------------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
echo "Logs will be collected in ""${TEST_SMOKE_OUTPUT_DIR}"


In smoke-test.sh line 45:
${PYTHON_VENV_DIR}/bin/behave --junit --junit-directory ${TEST_SMOKE_OUTPUT_DIR} \
^----------------^ SC2086: Double quote to prevent globbing and word splitting.
                                                        ^----------------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
"${PYTHON_VENV_DIR}"/bin/behave --junit --junit-directory "${TEST_SMOKE_OUTPUT_DIR}" \


In smoke-test.sh line 47:
                              smoke/features -D project_name=${TEST_NAMESPACE}                           
                                                             ^---------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
                              smoke/features -D project_name="${TEST_NAMESPACE}"                           


In smoke-test.sh line 48:
echo "Logs collected in "${TEST_SMOKE_OUTPUT_DIR}
                         ^----------------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
echo "Logs collected in ""${TEST_SMOKE_OUTPUT_DIR}"

For more information:
  https://www.shellcheck.net/wiki/SC2155 -- Declare and assign separately to ...
  https://www.shellcheck.net/wiki/SC3010 -- In POSIX sh, [[ ]] is undefined.
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```

Reply:
@gabemontero should we consider adding shellcheck to our `verify` test, perhaps as a follow-up PR?

Reply:
follow up PRs from anybody on the team would certainly be welcome for what you suggest @adambkaplan

Reply:
Done

Reply:
Used the shellcheck & updated the script and see nothing now.